README.md

```diff
@@ -0,0 +1,281 @@
+# browsertrix-harvester
+
+CLI app for performing a web crawl via [Browsertrix-Crawler](https://github.com/webrecorder/browsertrix-crawler), and optionally parsing structured metadata records from the crawl output [WACZ](https://replayweb.page/docs/wacz-format) file.
```

Comment:
That was a big part of the architecting discussions in [this Jira ticket](https://mitlibraries.atlassian.net/browse/TIMX-244) (sorry, should have linked in PR).  You pinpointed the big question going into this!

To summarize: the original proposal was to deploy the browsertrix-crawler as an ECS task with virtually no wrapper.  It would perform crawls, and because it had built-in S3 writing capabilities, theoretically we could write the WACZ file to S3.

One reason this fell down a bit: the crawler can write to S3, but only with a simple Access ID + Secret key approach.  Given that our context is role based, this was likely going to require the creation of special AWS CLI accounts just for this purpose.

This would also require another step in the TIMDEX pipeline to parse "raw" data like this into metadata records.  This could have taken many forms, but it would be another repository to manage and step to orchestrate.  This would be unlike OAI, that as a single step, gets data and puts it somewhere that Transmog can take from there.  

Ultimately, because there are not any other web crawling needs at the moment, and the _only_ web crawling need includes parsing metadata records, it was decided that combining them into a single app made sense.

By wrapping the crawler inside a container where we have a python CLI available, this gives us lots of flexibility on what we do _post_ crawl, which would not be otherwise possible.  If we ever wanted to extend this past crawl + harvest of metadata records, to just crawling, I think it'd be a CLI command away, with the added benefit of being able to write files, with more granular control, as needed.

Reply:
Ah, maybe I slightly misread your comment.  Do you mean more opinionated in the sense this app could _assume_ it's a) going to crawl the library website, and b) will parse metadata records? 

If so, even then, I'd argue it wouldn't make it much more opinionated.  In fact, a later PR will introduce a document that touches on just how opinionated it already is.  The `CrawlParser` class that is coming, is really geared towards parsing a crawl of the library website.  

My hope is that it hits a sweet spot where it could be extended to crawl and parse other websites if the need arose, but as-is, it's pretty flexible for crawl, parse, or crawl + parse of the library websites.

Reply:
Got it, that all makes sense!